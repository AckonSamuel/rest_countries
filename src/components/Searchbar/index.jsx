import PropTypes from 'prop-types';
import { useTheme } from '@mui/material/styles';
import Paper from '@mui/material/Paper';
import InputBase from '@mui/material/InputBase';
import IconButton from '@mui/material/IconButton';
import SearchIcon from '@mui/icons-material/Search';
import pxToRem from '../../assets/theme/pxToRem';

const Searchbar = ({ textListener, searchText }) => {
  const theme = useTheme();
  return (
    <Paper
      component="form"
      sx={{
        p: '2px 4px',
        display: 'flex',
        alignItems: 'center',
        maxWidth: 'inherit',
        marginTop: pxToRem(48),
        marginBottom: pxToRem(48),
        backgroundColor: theme.palette.background.primary,
        [theme.breakpoints.up('lg')]: {
          width: 430,
        },
        [theme.breakpoints.up('sm')]: {
          width: 343,
        },
      }}
    >
      <IconButton
        type="button"
        sx={{
          p: '10px',
          color: theme.palette.background.secondary,
        }}
        aria-label="search"
      >
        <SearchIcon />
      </IconButton>
      <InputBase
        sx={{ ml: 1, flex: 1 }}
        placeholder="Search Country"
        type="search"
        value={searchText}
        inputProps={{
          style: { color: theme.palette.background.secondary },
          'aria-label': 'search country',
          onChange: (e) => textListener(e.target.value),
        }}
      />
    </Paper>
  );
};

Searchbar.propTypes = {
  textListener: PropTypes.func.isRequired,
  searchText: PropTypes.string.isRequired,
};

export default Searchbar;
