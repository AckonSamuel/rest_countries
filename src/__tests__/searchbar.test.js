import React from 'react';
import renderer from 'react-test-renderer';
import Searchbar from '../components/Searchbar';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Searchbar />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
