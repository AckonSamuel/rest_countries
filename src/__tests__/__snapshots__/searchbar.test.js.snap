// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders searchbar component correctly 1`] = `
<form
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-pxp7br-MuiPaper-root"
>
  <button
    aria-label="search"
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="SearchIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
  </button>
  <div
    className="MuiInputBase-root MuiInputBase-colorPrimary css-84b1rr-MuiInputBase-root"
    onClick={[Function]}
  >
    <input
      aria-label="search country"
      className="MuiInputBase-input MuiInputBase-inputTypeSearch css-19fibzo-MuiInputBase-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search Country..."
      style={
        Object {
          "color": undefined,
        }
      }
      type="search"
      value="ghana"
    />
  </div>
</form>
`;
